src/scripts/scenes/falling.js

```diff
@@ -74,22 +75,58 @@ export default class FallingScene extends XrScene {
     const ballBody = new CANNON.Body({ mass: 1, material: this.objectMaterial });
     ballBody.addShape(this.ballShape);
     const material = new THREE.MeshPhongMaterial({ color: 'orange' });
-    const ball = new TriggerMesh(this.ballGeo, material);
+    const ball = new THREE.Mesh(this.ballGeo, material);
     ball.castShadow = true;
     ball.receiveShadow = true;
     this.world.addBody(ballBody);
 
-    ball.hover = function () {
-      if (!this.isSelected) {
-        this.material.color.set(0xFF0000);
+    let lastTime;
+    ball[Interactions] = {
+      hover_start() {
+        if (!this.isSelected) {
+          ball.material.color.set(0xFF0000);
+        }
+      },
+      hover_end() {
+        ball.material.color.set('orange');
+      },
```

Comment:
Is it possible to create a generalized `drag` and `drag_start` function for all objects so that one doesn't have to be written for each new object?

Reply:
Well, that would be great but I was trying to figure out how to do it.  Originally I wanted to have the object inherit from the generic drag/drop object that implemented all of those but then if you wanted to have the same be true for select_start and select_end and then again for  hover_start and hover_end, then the object might need multiple parents which isn't possible in Javascript.
Really, the defaults are just inside the interaction code and get run if the function isn't defined however to make it know that you want drag and drop, then you have to override at least one of the three drag and drop functions.  It's kinda yucky.  But not really worth overhauling I don't think.

Lastly, I have some semi generalized interaction things in pendulums that might interest you.  For example the yellow on hover and the drag with snapping points functions which create those interactions for the object you pass in:
https://github.com/Group-47-Capstone-2019/Immersive-Web-VR-AR/blob/f37515c4315210d16dceaf40dc20883697ce952c/src/scripts/scenes/pendulums.js#L17
https://github.com/Group-47-Capstone-2019/Immersive-Web-VR-AR/blob/f37515c4315210d16dceaf40dc20883697ce952c/src/scripts/scenes/pendulums.js#L133